lib/matplotlib/image.py

```diff
@@ -521,6 +521,8 @@ def contains(self, mouseevent):
 
     def write_png(self, fname):
         """Write the image to png file with fname"""
+        if self.origin == 'upper':
+            self._A = self._A[::-1]
```

Comment:
It is better to either flip it in the call `to_rgba` or flip the return `im` below.  Reassiging `_A` like this will cause the image to alternate orientation every time you called `write_png`.